packages/workflow/src/Workflow.ts

```diff
@@ -909,11 +909,10 @@ export class Workflow {
 			}
 		}
 
-		// Check if there is the actual "start" node
-		const startNodeType = 'n8n-nodes-base.start';
+		const startingNodeTypes = ['n8n-nodes-base.executeWorkflowTrigger', 'n8n-nodes-base.start'];
```

Comment:
One problem I see here is that for manual executions we might also choose `n8n-nodes-base.executeWorkflowTrigger` instead of `n8n-nodes-base.start`, whichever comes first.

My understanding is that for manual executions we should always begin from `n8n-nodes-base.start`, is this correct?

Reply:
My understanding is we are deprecating `n8n-nodes-base.start` throughout the app and enforcing the order `n8n-nodes-base.executeWorkflowTrigger` and then `n8n-nodes-base.start`, else throwing. But your comment made me realize that if a workflow contains these two node types, `n8n-nodes-base.start` was still being prioritized, so I added sorting before the check.

Reply:
Ok makes sense, and what about the new `manualTrigger`-type node?

Reply:
From the ticket I assumed manual is out of scope and so addressed later. I'll implement and ask David meanwhile.